lib/smart_proxy_remote_execution_ssh/api.rb

```diff
@@ -1,9 +1,208 @@
+require 'net/ssh'
+
+# When hijacking the socket of a TLS connection, we get a
+# OpenSSL::SSL::SSLSocket or Puma::MiniSSL::Socket, which don't behave
+# the same as real IO::Sockets.  We need to add recv and send for the
+# benefit of the Net::SSH::BufferedIo mixin, and closed? for our own
+# convenience.
+
+module Puma
+  module MiniSSL
```

Comment:
@mvollmer can you re-verify that this is working still for you? On my devel setup, with puma, I actually don't get to the cockpit instance with this (I was able to get there very occasionally). Webrick ssl or puma non-ssl works well, but the puma ssl seems like having some issues still.

Reply:
I haven't tested with Puma since a couple of weeks or so...